### PR TITLE
Site editor: fix padding and margin visualizer positioning

### LINKED ISSUE #41823: Padding visualizers do not work properly in the Site Editor
### Description

In the site editor blocks that use the placeholder illustration (see `PostFeaturedImage` and `Cover`) have an off-center bug for the dimensions visualizer.

### Step-by-step reproduction instructions

1. Use 13.5 RC1
2. Add a cover block
3. Set the cover block to use a featured image
4. Add padding and notice the visualizer is off

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/26996883/174665454-c9fc4dbd-1932-4ffc-870b-dff65877e629.mov

### Environment info

- GB 13.5RC1
- WP 6.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by talldan**
## What?
Fixes #41823

TLDR - Ensures `frameOffset` is not applied for popovers that render inline in the site editor's iframe.

Longer explanation:
The site editor has its content within an iframe, and usually most popovers (like the block toolbar) render in the main document. An x/y offset needs to be applied to these popovers so that they're in the correct position on the screen.

There are two situations where this offset doesn't need to be applied. If the popover and its anchor/reference are both not in the iframe (for example, a popover for the block inspector). Or if the popover and its anchor/reference are both in the iframe. The latter is the situation for the padding margin visualizers—they need to be rendered in the site editor's iframe so that they have access to the padding/margin CSS variables for content. But they were still incorrectly receiving the iframe offset.

## How?
Adds an extra check to see if the popover and its reference/anchor element are in the same document. This required gaining access to the floating element's owner document, which adds a little bit of extra code.

## Testing Instructions
1. Open the site editor
2. Insert a cover block
3. Adjust padding / margin
4. The visualizers should match the block position

### Before
Screen Shot 2022-09-08 at 4 30 22 pm

### After
Screen Shot 2022-09-08 at 4 29 51 pm

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Excellent, thank you for working on this! Here's a GIF showing the padding visualization looking correctly in both the post editor and the site editor:

offset

Code looks good, but would appreciate a sanity check. It seems like this did not have anything to do with the root padding property after all!

File: packages/components/src/popover/index.tsx [hunk 1]
```diff
@@ -416,10 +420,12 @@ const UnforwardedPopover = (
 	// document scrolls. Also update the frame offset if the view resizes.
 	useLayoutEffect( () => {
 		if (
-			// reference and floating are in the same document
+			// Reference and root documents are the same.
 			referenceOwnerDocument === document ||
-			// the reference's document has a view (i.e. window)
-			// and a frame element (ie. it's an iframe)
+			// Reference and floating are in the same document.
+			referenceOwnerDocument === floatingOwnerDocument ||
```
Comment (ntsekouras) [inline packages/components/src/popover/index.tsx]: Can't we just use the `refs.floating` from use `useFloating` hook?

Comment (talldan) [reply packages/components/src/popover/index.tsx]: That one's a callback ref.

Comment (ntsekouras) [reply packages/components/src/popover/index.tsx]: The `floating` is the callback ref, I meant from `refs` [property](https://floating-ui.com/docs/react-dom#internal-refs). I quickly tried it and seems to work fine. Didn't test thoroughly though.. Something like `referenceOwnerDocument === refs?.floating?.current?.ownerDocument`

Comment (talldan) [reply packages/components/src/popover/index.tsx]: Ah, who knew! (Nik did)

I'll update the PR.

File: packages/components/src/popover/index.tsx [hunk 2]
```diff
@@ -416,10 +416,12 @@ const UnforwardedPopover = (
 	// document scrolls. Also update the frame offset if the view resizes.
 	useLayoutEffect( () => {
 		if (
-			// reference and floating are in the same document
+			// Reference and root documents are the same.
 			referenceOwnerDocument === document ||
-			// the reference's document has a view (i.e. window)
-			// and a frame element (ie. it's an iframe)
+			// Reference and floating are in the same document.
+			referenceOwnerDocument === refs?.floating?.current?.ownerDocument ||
```
Comment (ntsekouras) [inline packages/components/src/popover/index.tsx:422]: Do we need to add this to the deps or having `update` there is enough? It seems to work well even without being in deps..

**Review by ntsekouras — APPROVED** [review]
Thanks Dan! 💯

File: packages/components/src/popover/index.tsx (hunk 2 above)
Comment (talldan) [reply packages/components/src/popover/index.tsx:422]: Thanks for reviewing.

Because it's a ref I don't think it needs to be, and I don't think it'll be a value that can change 🤔 

I also didn't notice any issues, so I think lets leave it as it is for now.

**Comment by ciampo** [comment]
Thank you Dan 🙏
